scripts/portfolio.js

```diff
@@ -18,10 +28,58 @@ Project.prototype.toHtml = function() {
   return html;
 };
 
-myProjectData.forEach(function(ele) {
-  projects.push(new Project(ele));
-});
+ProjectWip.prototype.toHtml = function() {
+  // use Handlebar templating to create project data for HTML.
+  var source = $('#wip-template').html();
+  var template = Handlebars.compile(source);
+  var html = template(this);
+
+  return html;
+};
+
+
+// loadAll will take ALL the parsed JSON data and put it in the project arrays for processing.
+Project.loadAll = function(dataWePassIn) {
+  dataWePassIn.forEach(function(ele) {
+    if (ele.prjType === 'PROJECT') {
+      Project.all.push(new Project(ele));
+    } else {
+    //if (ele.prjType === 'WIP') {    <<---- this is the other type. Allows for more types...?
+      ProjectWip.all.push(new ProjectWip(ele));
+    };
+  });
+};
+
+// This function below will retrieve ALL the data from either a local or remote
+// source, process it, then hand off control to the View...
+
+Project.fetchAll = function() {
+  if (localStorage.portfolioData) {
+      // When our data is already in localStorage:
+      // 1. We can process it,
+      // 2. Then we can render the index page.
+    var storedData = JSON.parse(localStorage.getItem('portfolioData'));
+    Project.loadAll(storedData);
+  } else {
+      //  1. Load our json data,
+      //  2. Store that data in localStorage so we can skip the server call next time.
+      //  3. And then render the index page. */
+    $.ajax('/data/portfolioData.json', {
+      method: 'GET',
+      success: successHandler,
+      error: errorHandler
+    });
+  }
+};
+
+function successHandler(data) {
+  localStorage.setItem('portfolioData', JSON.stringify(data));
+  Project.loadAll(data);
+  portfolioView.renderIndexPage();
```

Comment:
There's a typo here. When data is first loaded, it doesn't render to the page and an error appears in the web console. On reloading you won't see the error. It helps to periodically clear your localStorage to catch bugs like that.